src/Microsoft.HttpRepl/HttpState.cs

```diff
@@ -84,10 +88,16 @@ public IEnumerable<string> GetApplicableContentTypes(string method, string path)
 
         public Uri GetEffectivePath(string commandSpecifiedPath)
         {
-            return GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath);
+            return GetEffectivePath(BaseAddress, string.Join('/', PathSections.Reverse()), commandSpecifiedPath, null);
+        }
+
+
+        public Uri GetEffectivePathWithQueryString(string commandSpecifiedPath, Dictionary<string,IEnumerable<string>> queryString)
```

Comment:
I need to give this more thought. I _think_ its more correct to include the query string by default (and force the use of the static if you want only some of the parts - as is done in `GetEffectivePathForPrompt`). But I need to dig through some of the usages first.